src/OAuth/OAuth2/Service/BattleNetBase.php

```diff
@@ -0,0 +1,81 @@
+<?php
+namespace OAuth\OAuth2\Service;
+
+use OAuth\OAuth2\Token\StdOAuth2Token;
+use OAuth\Common\Http\Exception\TokenResponseException;
+use OAuth\Common\Http\Uri\Uri;
+use OAuth\Common\Consumer\Credentials;
+use OAuth\Common\Http\Client\ClientInterface;
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Http\Uri\UriInterface;
+
+class BattleNetBase extends AbstractService
+{
+    const SCOPE_WOW_PROFILE				= 'wow.profile';
+    const SCOPE_SC2_PROFILE				= 'sc2.profile';
+
+    /**
+    * BattleNet Region
+    *
+    * @var \region
+    */
+    protected $region;
+
+    public function __construct(Credentials $credentials, ClientInterface $httpClient, TokenStorageInterface $storage, $scopes = array(), UriInterface $baseApiUri = null)
```

Comment:
This was actually a copy/paste from an existing file in the library that I modified as needed.  I agree about the line length and it looks like it has been fixed in other files since then.